src/ivalue_manager.rs

```diff
@@ -247,25 +337,24 @@ impl<'a> WriteHolder<IValue, IValue> for IValueKeyHolderWrite<'a> {
                     o.insert(key.to_string(), v.take());
                 }
             }
-            let val = root.take();
-            self.set_root(Some(val))?;
         } else {
-            update(&path, self.get_value().unwrap().unwrap(), |mut val| {
-                if let Some(o) = val.as_object_mut() {
+            update_in_closure(&path, self.get_value().unwrap().unwrap(), |val| {
```

Comment:
I guess this allows you to update an existing value, maybe call one update and the other replace?

Reply:
What about `update` and `calculate`?
(the caller is doing the actual replacement)

Reply:
How come the caller doing the replacement? The caller returns a new value and the update functions replace it no?

Reply:
Will rename to `update` and `replace`